### PR TITLE
[Algorithms] Select k-th element & Median selecting

### PR DESCRIPTION
**Pull request by Tismoney**

### CODE REVIEW

File: sorting/select_k/select_k.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            return select_k(left, k)
+        elif k < len(left) + len(pivots):
+            return pivot
+        else:
+            return select_k(right, k - len(left) - len(pivots))
+
+
+def split_array(arr, pivot):
+    """
+    Function of split array into three: more, less and equally than pivot
```
Comment (gkorepanov) [inline sorting/select_k/select_k.py]: Would be nice to fix grammar

File: sorting/select_k/select_k.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+def select_k(arr, k):
```
Comment (gkorepanov) [inline sorting/select_k/select_k.py]: Would be great to add asymptotic complexity (O(?)) to docstrings of non-trivial functions (e.g. `select_k`, `split_array`)

File: sorting/select_k/select_k.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    for element in arr:
+        if element < pivot:
+            left.append(element)
+        elif element > pivot:
+            right.append(element)
+        else:
+            pivots.append(element)
+    return left, right, pivots
+
+def select_median(arr):
```
Comment (gkorepanov) [inline sorting/select_k/select_k.py]: is this linear (O(n)) worst-case or something else?

File: sorting/select_k/select_k.py (hunk 2 above)
Comment (gkorepanov) [reply sorting/select_k/select_k.py]: Also would be great to give explanation general comments for non-trivial algorithms

Comment (gkorepanov) [reply sorting/select_k/select_k.py]: This seems like to have expected linear time?

File: sorting/select_k/select_k.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+def select_k(arr, k):
+    """
+    Function of selecting k-th element of array
+    Parameters
+    ----------
+        arr:    list, input array
+        k:      int, index number in sorted array
+        
+    Examples
```
Comment (gkorepanov) [inline sorting/select_k/select_k.py:11]: Great examples sections!

File: sorting/select_k/select_k.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    --------
+        >>> arr = [1, 9, 2, 8, 3, 7, 4, 6, 5]
+        >>> select_k(arr, 1)
+        2
+        >>> select_k(arr, -1)
+        9
+        >>> select_k(arr, 5) == sorted(arr)[5]
+        True
+    """
+    k = len(arr) + k if k < 0 else k
```
Comment (gkorepanov) [inline sorting/select_k/select_k.py:25]: where negative k values are used?

Comment (Tismoney) [reply sorting/select_k/select_k.py:25]: It need if you want to select last element. In sorted array you just use `-1` index instead `len(arr) - 1`.

File: sorting/select_k/select_k.py (hunk 3 above)
Comment (Tismoney) [reply sorting/select_k/select_k.py]: Honestly, O(n) is average complexity. In worst-case it actually has O(n^2). This approval is based on this [page](https://rcoh.me/posts/linear-time-median-finding/).

**Pull request merged**